addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/fritzaha/handler/FritzAhaBridgeHandler.java

```diff
@@ -0,0 +1,282 @@
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.fritzaha.FritzAhaBindingConstants;
+import org.openhab.binding.fritzaha.config.FritzAhaBridgeConfiguration;
+import org.openhab.binding.fritzaha.internal.discovery.FritzBoxDiscoveryService;
+import org.openhab.binding.fritzaha.internal.hardware.FritzahaWebInterface;
+import org.openhab.binding.fritzaha.internal.hardware.callbacks.FritzAhaDiscoveryCallback;
+import org.openhab.binding.fritzaha.internal.hardware.callbacks.FritzAhaUpdateXmlCallback;
+import org.openhab.binding.fritzaha.internal.model.DeviceModel;
+import org.openhab.binding.fritzaha.internal.model.SwitchModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler for a FRITZ!Box device. Handles polling of values from AHA devices.
+ * 
+ * @author Robert Bausdorf
+ * @since 2.0
+ */
+public class FritzAhaBridgeHandler extends BaseBridgeHandler {
+	/**
+	 * Logger
+	 */
+	private static Logger LOG = LoggerFactory
+			.getLogger(FritzAhaBridgeHandler.class);
+	/**
+	 * the refresh interval which is used to poll values from the fritzaha.
+	 * server (optional, defaults to 10 s)
+	 */
+	private long refreshInterval = 10;
+	/**
+	 * Interface object for querying the FRITZ!Box web interface
+	 */
+	private FritzahaWebInterface connection;
+	/**
+	 * Holder for last data received from the box.
+	 */
+	private Map<String, DeviceModel> deviceList;
+	/**
+	 * Job which will do the FRITZ!Box polling
+	 */
+	private FritzPolling pollingRunnable;
+	/**
+	 * Schedule for polling
+	 */
+	private ScheduledFuture<?> pollingJob;
+
+	/**
+	 * Constructor
+	 * 
+	 * @param bridge
+	 *            Bridge object representing a FRITZ!Box
+	 */
+	public FritzAhaBridgeHandler(Bridge bridge) {
+		super(bridge);
+		this.deviceList = new TreeMap<String, DeviceModel>();
+		this.pollingRunnable = new FritzPolling(this);
+	}
+
+	/**
+	 * Initializes the bridge.
+	 */
+	@Override
+	public void initialize() {
+		LOG.debug("About to initialize bridge "
+				+ FritzAhaBindingConstants.BRIDGE_FRITZAHA);
+		LOG.debug("lookup discovered boxes");
+		Bridge bridge = this.getThing();
+		FritzAhaBridgeConfiguration config = null;
+		if (bridge != null) {
+			String discoveredIp = bridge.getConfiguration().getProperties()
+					.get(FritzAhaBindingConstants.IP_ADDRESS).toString();
+			config = this.getConfigAs(FritzAhaBridgeConfiguration.class);
+			config.setIpAddress(discoveredIp);
+			LOG.info("discovered fritzaha bridge initialized: "
+					+ config.toString());
+		} else {
+			config = this.getConfigAs(FritzAhaBridgeConfiguration.class);
+			LOG.info("default fritzaha bridge initialized: "
+					+ config.toString());
+		}
+		this.refreshInterval = config.getPollingInterval();
+		this.connection = new FritzahaWebInterface(config);
+		if (config.getPassword() != null) {
+			this.onUpdate();
+		} else {
+			LOG.warn("no password set - no polling");
```

Comment:
IF the password is really required, you should not log a warning here, but set the thing status to OFFLINE-CONFIGURATION_ERROR (see https://github.com/eclipse/smarthome/blob/master/docs/sources/architecture/thing-status.md#status-detail-mapping)